### PR TITLE
Describe live variable analysis in developer docs

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: docs/source/developer/live_variable_analysis.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+end of the current block).  This means variables can be released earlier than in
+regular Python code.
+
+The behavior of the live variable analysis affects memory usage of the compiled
+code.  Internally, Numba does not differentiate temporary variables and user
+variables.  Since each operation generates at least one temporary variable,
+a function can accumulate a high number of temporary variables if they are not
+released as soon as possible.
+Our generator implementation can benefit from early releasing of variables,
+which reduces the size of the state to suspend at each yeild point.
```
Comment (pitrou) [inline docs/source/developer/live_variable_analysis.rst]: s/yeild/yield/

**Comment by seibert** [comment]
Not going to wait for Travis-CI to test a typo fix.  :)

**Comment by Juanlu001** [comment]
Just for the record: you can add `[skip ci]` anywhere in the commit message to avoid triggering a continuous build https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build

**Comment by seibert** [comment]
Oh, nice!  Good to know.

File: docs/source/developer/live_variable_analysis.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+(Releated issue https://github.com/numba/numba/pull/1611)
+
+Numba uses reference-counting for garbage collection, a technique that
+requires cooperation by the compiler.  The Numba IR encodes the location
+where a decref must be inserted.  These locations are determined by live
+variable analysis.  The corresponding source code is the ``_insert_var_dels()``
+method in https://github.com/numba/numba/blob/master/numba/interpreter.py.
+
+
+In Python semantic, once a variable is defined inside a function, it is alive
```
Comment (ahmadia) [inline docs/source/developer/live_variable_analysis.rst]: --semantic-- -> semantics

**Comment by ahmadia** [comment]
Ooops, missed the window on this, nice work!

**Pull request merged**